src/libs/antares/study/study.cpp

```diff
@@ -145,6 +146,11 @@ void Study::clear()
     gotFatalError = false;
 }
 
+void Study::setWriter(Solver::IResultWriter::Ptr writer)
```

Comment:
Why this function ?
I said in a previous review (#794 : src/solver/main/economy.cpp) that the writer should be created in the Application class, where it is first needed, and then passed to any component that needs it (simulation, xcast, ...)
Now, I would go a bit further : the writer should be created in the study as **study->parameters** has information to do it, and then it should be passed to Application, simulation, and so on.